code/iaas/allocator/src/main/java/io/cattle/platform/allocator/service/impl/AllocatorServiceImpl.java

```diff
@@ -348,6 +350,17 @@ protected LockDefinition getInstanceLockDef(Instance origInstance, List<Instance
         for (Volume v : volsToLock) {
                 locks.add(new AllocateConstraintLock(AllocateConstraintLock.Type.VOLUME, v.getId().toString()));
         }
+        
+        List<Volume> volumes = InstanceHelpers.extractVolumesFromMounts(origInstance, objectManager);
+        for (Volume volume: volumes) {
+            StorageDriver driver = objectManager.loadResource(StorageDriver.class, volume.getStorageDriverId());
+            String accessMode = DataAccessor.fieldString(driver, StorageDriverConstants.FIELD_VOLUME_ACCESS_MODE);
+            if (accessMode.equals(VolumeConstants.ACCESS_MODE_SINGLE_HOST_RW) || accessMode.equals(VolumeConstants.ACCESS_MODE_SINGLE_INSTANCE_RW)) {
+                if (!volume.getAllocationState().equals("active")) {
```

Comment:
Can you switch to CommonStates.ACTIVE, also as a little trick in java when doing string compare against constants reverse the expression.  So

```java
CommonState.ACTIVE.equals(volume.getAllocationState())
```
Doing that you never have to worry about a null pointer exception.